numpy/core/src/multiarray/ctors.c

```diff
@@ -1862,9 +1869,15 @@ PyArray_CheckFromAny(PyObject *op, PyArray_Descr *descr, int min_depth,
     if (obj == NULL) {
         return NULL;
     }
+
     if ((requires & NPY_ARRAY_ELEMENTSTRIDES) &&
         !PyArray_ElementStrides(obj)) {
         PyObject *ret;
+        if( requires & _NPY_ARRAY_ENSURENOCOPY ) {
+            PyErr_SetString(PyExc_ValueError,
+                            "Unable to avoid copy while creating a new array.");
+            return NULL;
+        }
```

Comment:
Can you remove the check in line 1876, run the test suite to see which tests fail, and reproduce them with `_NPY_ARRAY_ENSURENOCOPY` ?

Reply:
Same here. I will do that but curious to know the motivation.

Reply:
the same motivation as any coverage check: to make sure it works as expected and that future code will not break it.

Reply:
Coverage reported the code path as "not covered"

Reply:
The test I added for this is at https://github.com/numpy/numpy/blob/d9a9785ada0b4a2b8694511dd854121f618f4f56/numpy/core/tests/test_multiarray.py#L8007

Reply:
I think that codecov didn’t update the status because one test failed. May be it only updates when the full build is successfull?

Reply:
No, it is uploaded fine, this branch is not covered since you never test `NPY_ARRAY_ELEMENTSTRIDES`.

Reply:
I tried several example to hit this line. In case of scalars this line is never hit because, `PyArray_ElementStrides` is never `false`. In case of lists, tuples, arrays, the error is already raised in `PyArray_FromAny` (call to which is made just a few lines above).

Reply:
It seems the check on line 1873-1874 was added in the far past (2011, when the code , and cannot be hit. Could you make it unconditionally raise with a "cannot be reached" error instead of the `PyArray_NewCopy()`, and run it through CI to make sure our analysis is correct?